src/Microsoft.AspNet.Identity/SignInManager.cs

```diff
@@ -41,7 +42,7 @@ public class SignInManager<TUser> where TUser : class
             {
                 throw new ArgumentNullException(nameof(userManager));
             }
-            if (contextAccessor == null || contextAccessor.HttpContext == null)
```

Comment:
Won't most of the methods in this class now NullRef in `contextAccessor.HttpContext` is null? I think it is better to throw in the ctor? Maybe it shouldn't `ArgumentNullException` but instead an `InvalidArgumentExeption` with a clear explanation that the `HttpContext` property can't be null.